phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -898,9 +898,9 @@ public MetaDataMutationResult updateCache(PName tenantId, List<String> functionN
      * @return true if the PTable contained by result was modified and false otherwise
      * @throws SQLException if the physical table cannot be found
      */
-    private boolean addColumnsAndIndexesFromAncestors(MetaDataMutationResult result, Long resolvedTimestamp,
-                                                      boolean alwaysAddAncestorColumnsAndIndexes,
-                                                      boolean alwaysHitServerForAncestors)
+    private boolean addColumnsIndexesAndLastDDLTimestampsFromAncestors(MetaDataMutationResult result, Long resolvedTimestamp,
```

Comment:
I believe we should refactor this function a little so that it works for views, view indexes and regular indexes. This way there is only one place in the code where we are adding information from the ancestors. For view/view index we add columns, indexes, lastddltimestamp. For indexes we just add lastddltimestamp.

Reply:
We could rename this function to something like `addMetadataInformationFromAncestors` and then add a javadoc explaining what metdata we need from ancestors depending upon the table type.

Reply:
> This way there is only one place in the code where we are adding information from the ancestors.
@tkhurana What if updateCache is not called for the index table directly? If the client resolves just the base table for a query, this method will not execute for the index, right? One thing we could do is to create a new PTable in the case of tables with a new list of indexes inside it which have the ancestor map.

Reply:
How does it handle view indexes in that case @palashc ? I am assuming if this method is called for all types of index.

Reply:
Good point. I just verified with a test that a new client will not be able to populate the map for both indexes and view indexes. I created a table, view and view-index using client-1. Then I queried the view using client-2. In client2's cache, view index is not populated with the map.

Reply:
> I am assuming this method is called for all types of index.

Yes, but I think it is called only when the index is directly being used in the plan. Thus, we might have to populate the map for indexes when we add an index to the cache in `PMetaDataImpl.addTable`